### PR TITLE
Тюрмина А.Н. 381806-3 ЛР2

### PR DESCRIPTION
**Pull request by sashatyurmina**
### Pull Request Readiness Checklist

See details at [code review]( https://github.com/UNN-ITMM-Software/devtools-course-practice/blob/main/lab-guide/lab-1-setup.md#%D1%80%D0%B5%D1%86%D0%B5%D0%BD%D0%B7%D0%B8%D1%8F-%D0%BA%D0%BE%D0%B4%D0%B0)

- [ ] Code is well documented and more information can be seen on the [website](https://devtools-course-practice-2021.readthedocs.io/en/latest/modules.html) (recommended)
- [x] Make sure that the testing has passed successfully on [GitHub Actions](https://github.com/UNN-ITMM-Software/devtools-course-practice/actions)
- [x] Get two approve from your fellow students
- [x] Take a review from one of the administrators

### CODE REVIEW

**Comment by sashatyurmina** [comment]
@dasha-ludina

**Comment by sashatyurmina** [comment]
@shekotilova-yuliya

**Comment by sashatyurmina** [comment]
@dasha-ludina 
@shekotilova-yuliya

File: modules/book_discount/src/calculate_price.cpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
+// Copyright 2021 Tyurmina Alexandra
+
+#include "include/calculate_price.h"
+#include <stdio.h>
```
Comment (dasha-ludina) [inline modules/book_discount/src/calculate_price.cpp:4]: Необходим пробел между 3 и 4 строками

File: modules/book_discount/test/test_calc_price.cpp [hunk 2]
```diff
@@ -0,0 +1,91 @@
+// Copyright 2021 Tyurmina Alexandra
+
+#include <gtest/gtest.h>
+#include "include/calculate_price.h"
```
Comment (dasha-ludina) [inline modules/book_discount/test/test_calc_price.cpp:4]: Необходим пробел между 3 и 4 строками

File: modules/book_discount/test/test_calc_price.cpp [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+TEST(CalculatePrice, Can_Create_Basket_With_Zero_Value) {
+  ASSERT_NO_THROW(CalculatePrice(0, 0, 0, 0, 0));
+}
+
+TEST(CalculatePrice,
+  Calculate_Price_Same_Parts_Of_Books_Without_Discount) {
+  CalculatePrice b(7, 0, 0, 0, 0);
+  double price = 56.0;
+
+  EXPECT_EQ(price, b.TotalSum());
```
Comment (shekotilova-yuliya) [inline modules/book_discount/test/test_calc_price.cpp:20]: лучше использовать ASSERT вместо EXPECT

File: modules/book_discount/test/test_calc_price.cpp [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+TEST(CalculatePrice,
+  Calculate_Price_With_Discount_Only_For_Different_Books) {
+  CalculatePrice b(7, 1, 0, 0, 0);
+  double price = 63.2;
+
+  EXPECT_EQ(price, b.TotalSum());
```
Comment (shekotilova-yuliya) [inline modules/book_discount/test/test_calc_price.cpp:28]: тут и далее аналогично

**Comment by dkurt** [comment]
please push some commit to check the codecov once again

**Comment by sashatyurmina** [comment]
@dasha-ludina @shekotilova-yuliya исправила

**Review by shekotilova-yuliya — APPROVED** [review]

**Review by dasha-ludina — APPROVED** [review]
Все хорошо)

**Comment by dkurt** [comment]
Seems like the problem with codecov is related to https://github.com/UNN-ITMM-Software/devtools-course-practice/issues/119. The thing is that CI compares PR changes in the proposed branch. But should rebase with actual master first.

You can solve it by rebasing local branch before we solved it

image

**Comment by WerWebWer** [comment]
Выполните следующее
``` git
git remote add upstream https://github.com/UNN-ITMM-Software/devtools-course-practice
git checkout main
git pull upstream main
git rebase main branch_lab3
git push upstream -f branch_lab3
```

**Comment by sashatyurmina** [comment]
@WerWebWer готово

**Comment by WerWebWer** [comment]
Как на счёт параметризированных тестов? Для того чтобы объединить похожие тесты

**Review by WerWebWer — APPROVED** [review]

**Pull request merged**